doc/source/development/rfc/rfc104_gdal_cli.rst

```diff
@@ -0,0 +1,1175 @@
+          "description":"Creation option",
+          "min_count":0,
+          "max_count":2147483647,
+          "category":"Base"
+        },
+        {
+          "name":"layer-creation-option",
+          "type":"string_list",
+          "description":"Layer creation option",
+          "min_count":0,
+          "max_count":2147483647,
+          "category":"Base"
+        },
+        {
+          "name":"overwrite",
+          "type":"boolean",
+          "description":"Whether overwriting existing output is allowed",
+          "default":false,
+          "min_count":0,
+          "max_count":1,
+          "category":"Base"
+        },
+        {
+          "name":"update",
+          "type":"boolean",
+          "description":"Whether updating existing dataset is allowed",
+          "default":false,
+          "min_count":0,
+          "max_count":1,
+          "category":"Base"
+        },
+        {
+          "name":"overwrite-layer",
+          "type":"boolean",
+          "description":"Whether overwriting existing layer is allowed",
+          "default":false,
+          "min_count":0,
+          "max_count":1,
+          "category":"Base"
+        },
+        {
+          "name":"append",
+          "type":"boolean",
+          "description":"Whether appending to existing layer is allowed",
+          "default":false,
+          "min_count":0,
+          "max_count":1,
+          "category":"Base"
+        },
+        {
+          "name":"input-layer",
+          "type":"string_list",
+          "description":"Input layer name(s)",
+          "min_count":0,
+          "max_count":2147483647,
+          "category":"Base"
+        },
+        {
+          "name":"output-layer",
+          "type":"string",
+          "description":"Output layer name",
+          "min_count":0,
+          "max_count":1,
+          "category":"Base"
+        }
+      ],
+      "output_arguments":[
+      ],
+      "input_output_arguments":[
+        {
+          "name":"output",
+          "type":"dataset",
+          "description":"Output vector dataset",
+          "min_count":1,
+          "max_count":1,
+          "category":"Base",
+          "dataset_type":[
+            "vector"
+          ],
+          "input_flags":[
+            "name",
+            "dataset"
+          ],
+          "output_flags":[
+            "dataset"
+          ]
+        }
+      ]
+    }
+
+.. _rfc104_gdal_vector_pipeline_examples:
+
+A few invocations of "gdal vector pipeline"
++++++++++++++++++++++++++++++++++++++++++++
+
+  .. code-block:: shell
+
+     # The use of the '!' as a step separator is to prevent Unix or Windows shells from
+     # trying to use other processes for the "reproject" or "write" steps.
+     # Below is a single-process pipeline.
+     $ gdal vector pipeline read poly.gpkg ! reproject --dst-crs=EPSG:4326 ! write out.parquet --overwrite
+
+     # Alternative without the "vector" and "pipeline" subcommands, and with --progress
```

Comment:
I would either always have `pipeline` or never. But not that many options. Many options for the same task are confusing.